Utility/envelope.cpp

```diff
@@ -0,0 +1,71 @@
+/******************************************************************************
+/ envelope.cpp
+/
+/ Copyright (c) 2019
+/
+/ Permission is hereby granted, free of charge, to any person obtaining a copy
+/ of this software and associated documentation files (the "Software"), to deal
+/ in the Software without restriction, including without limitation the rights to
+/ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+/ of the Software, and to permit persons to whom the Software is furnished to
+/ do so, subject to the following conditions:
+/
+/ The above copyright notice and this permission notice shall be included in all
+/ copies or substantial portions of the Software.
+/
+/ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+/ EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+/ OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+/ NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+/ HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+/ WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+/ FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+/ OTHER DEALINGS IN THE SOFTWARE.
+/
+******************************************************************************/
+
+#include "stdafx.h"
+#include "../reaper/localize.h"
+#include "envelope.hpp"
+
+namespace envelope
+{
+	// wraps GetEnvelopeStateChunk() to make it more save
+	// until maybe it gets enhanced one day
+	// https://forum.cockos.com/showthread.php?p=2142245#post2142245
+	// throws envUtil::bad_get_env_chunk_big on failure
+	std::string GetEnvelopeStateChunkBig(TrackEnvelope* envelope, bool isUndo /*= false*/)
```

Comment:
You can save an indentation level in this file using `std::string envelope::GetEnvelopeStateChunkBig` in this line and removing the namespace (and no need to repeat the namespace name within the function).

Reply:
But then every further declared function would have to use envelope:: also, no?
Is the repetitiveness worth saving the indentation level you think? :)

Reply:
It's the same as when defining class member functions.

Reply:
Point taken, I'll remove the namespace { }.